src/App.tsx

```diff
@@ -1,24 +1,336 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Error } from './components/Error';
+import { Todo } from './types/Todo';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Filters } from './types/Filters';
 import { UserWarning } from './UserWarning';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 11689;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Filters.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isPageLoading, setIsPageLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdatingId, setIsUpdatingId] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [editQuery, setEditQuery] = useState('');
+  const [editQueryPrev, setEditQueryPrev] = useState('');
+
+  const handleErrorSet = (errMessage: string) => {
+    setErrorMessage(errMessage);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const closeError = () => {
+    setErrorMessage('');
+  };
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+      setIsPageLoading(true);
```

Comment:
you should do it before you load todos, also you should eventually turn it to false when todos are loaded